### PR TITLE
Middle_Element_Of_Singly_Linked_List_C++.md

### LINKED ISSUE #7260: Middle Of a Singly Linked List [Documentation]
### Description

I will be explaining the concept behind this question with proper code.

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by ritika728**
<hr>

## Description 📜

I have explained how to find the middle element in a singly linked list with proper code and examples.
Fixes #7260 

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [x] C/CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

<hr>

### CODE REVIEW

**Comment by ritika168** [comment]
@apeksha235 @Saviour1001 @Suvraneel please review!

**Comment by Suvraneel** [comment]
@gunjan-g

File: C_CPP/Data Structures/Linked List/Singly Linked List (Normal Linked List)/Middle_Element_Of_Singly_Linked_List_C++.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
+# MIDDLE ELEMENT OF SINGLY LINKED LIST
+## INPUT :
+A singly linked list which is inputed by user.
```
Comment (vasu-1) [inline C_CPP/Data Structures/Linked List/Singly Linked List (Normal Linked List)/Middle_Element_Of_Singly_Linked_List_C++.md:3]:
```suggestion
Singly linked lists which are inputed by user.  
```

File: C_CPP/Data Structures/Linked List/Singly Linked List (Normal Linked List)/Middle_Element_Of_Singly_Linked_List_C++.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+A singly linked list which is inputed by user.  
+Example 1 : 10->9->8->7->6  
+Example 2 : 10->9->8->7->6->5
+## OUTPUT:
+Middle of the given linked list. If list has even number of nodes, then print the second middle element.  
+For first example, middle will be 8.  
+For second example, middle will be 7. (because we will take the second middle element in case of even number of nodes)
+## APPROACH :
+* Ask the user for number of nodes in the linked list.
+* Create the Linked list by inputing the data value given by user.
```
Comment (vasu-1) [inline C_CPP/Data Structures/Linked List/Singly Linked List (Normal Linked List)/Middle_Element_Of_Singly_Linked_List_C++.md:12]:
```suggestion
* Create the Linked list by inputting the data value given by user.
```

File: C_CPP/Data Structures/Linked List/Singly Linked List (Normal Linked List)/Middle_Element_Of_Singly_Linked_List_C++.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+* Traverse the whole linked list and calculate the number of nodes in that list. Lets say : n nodes are there.
```
Comment (vasu-1) [inline C_CPP/Data Structures/Linked List/Singly Linked List (Normal Linked List)/Middle_Element_Of_Singly_Linked_List_C++.md:13]:
```suggestion
* Traverse the whole linked list and calculate the number of nodes in that list. Let's say : n nodes are there.
```

**Review by vasu-1 — CHANGES_REQUESTED** [review]
Changes Required !

Do comment in the code !

Happy Diwali !!

**Comment by ritika728** [comment]
Happy Diwali! 🪔
@vasu-1 I have made the required changes.

**Comment by Suvraneel** [comment]
@ritika728  LGTM 👍🏻  💯

**Pull request merged**